Mathlib/FieldTheory/CardinalEmb.lean

```diff
@@ -0,0 +1,342 @@
+/-
+Copyright (c) 2024 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import Mathlib.FieldTheory.SeparableClosure
+import Mathlib.FieldTheory.PurelyInseparable
+import Mathlib.LinearAlgebra.FreeAlgebra
+import Mathlib.Order.Interval.Set.WithBotTop
+import Mathlib.Order.DirectedInverseSystem
+
+/-!
+# Number of embeddings of an algebraic extension of infinite separable degree
+
+We show that if `E/F` is an algebraic extension of infinite separable degree, then
+`#(Field.Emb F E) = 2 ^ Field.sepDegree F E`. If `E/F` is moreover separable, then
+`#(Field.Emb F E) = 2 ^ Module.rank F E`. This is in contrast to the case of finite
+separable degree, where `#(Field.Emb F E) = Module.rank F E`.
+
+We use a transfinite recursive construction that is fairly standard in set theory, but the author
```

Comment:
```suggestion
## Implementation details

We use a transfinite recursive construction that is fairly standard in set theory, but the author
```
Or maybe "Sketch of the proof", or something like that.

Could you please also add a `## Main results` where you give the name of the result?